### ISSUE TITLE
[event] Website Sprint + OSCD@HacktoberFest

### ISSUE THREAD

**Issue by emmyft** [osc-delft/ideas-and-plans#1]
**One line summary: from now to October 31, let's work together to launch v1 of the new OSCD website!** 🚀 

Two modes of participation:
1. Asynchronously, i.e. whenever is good for you
2. Synchronously work together:
   - Date: October 30 (Friday)
   - Time: 9:30am-12noon CET
   - Venue: Zoom (link to be shared via email)

# What is this?
We recognise the need for a new website for the OSCD community - the [Data Champions website](https://www.tudelft.nl/en/library/current-topics/research-data-management/r/support/data-champions/) is very nice to look at, but difficult to maintain, modify and scale.

The idea is to have a new, GitHub pages based website where it not only displays information but also acts as a way for the community members to interact with each other and build things that we want to build, together 🤝  

Currently, the website sits in the [osc-delft.github.io](https://github.com/osc-delft/osc-delft.github.io) repository - you can see it rendered at https://osc-delft.github.io.

I've built a frame for the website by adopting the one from the [Open Life Science programme](https://openlifesci.org) - but it needs further work before it can be launched to the community.

**Goal: Complete the website to the point where it's ready to be launched to the community and public** 🎉 

# What would I do in this Sprint? Why should I join?
You can:
- learn about what GitHub is, how it can be used for openly collaborating on projects - and then immediately use what you've learnt! 💯 
- get to know and work together with other members of the community 
- contribute to building this online community home 🏡 
- participate in one of the biggest open-source events in the world - Hacktoberfest! (see below)

[**Hacktoberfest**](https://hacktoberfest.digitalocean.com/) is a global event happening in the month of October to drive more people to contribute to open-source. If you make 4 pull requests (PRs) to a participating repository in October and they are merged (accepted), you have the chance to win a free t-shirt 👕  (or, plant a tree)

For all PRs made to any of the osc-delft repositories (not only in the website repo), I'll make them `hacktoberfest-accepted` when they are merged, so that they will count towards your Hacktoberfest participation. 

**I have no idea where to start contributing** - we've got you! I'll mark issues with a `hacktoberfest` label to help you find the ones that are likely to result in a PR. The [Contribution guidelines](https://github.com/osc-delft/osc-delft.github.io/blob/master/CONTRIBUTING.md) should help you find ways to contribute too

**I don't code...** We have plenty of work to do on the text, graphics and documentation of the website! Whether it's fixing typos/broken links, or suggestion new ideas/pages, we need your help

**I can't make it to the actual Sprint** 😢 Don't worry- you're still more than welcomed to contribute and make PRs at a time convenient for you! Just remember, if you'd like those PRs to be counted towards Hacktoberfest, ideally they should be made before October 31 (so we have some time to review and merge them too) (see the Asynchronous section below)

# Asynchronous participation - anytime, anywhere
Start by having a look at the [list of issues](https://github.com/osc-delft/osc-delft.github.io/issues) and the [contributing guidelines](https://github.com/osc-delft/osc-delft.github.io/blob/master/CONTRIBUTING.md) 👀 

You can:
- help with an existing issue - by commenting and/or helping complete the task(s) described
- open a new issue pointing up errors/bugs you see, or improvements! 

# Sprint together 🏃 - Schedule on October 30
*You don't have to join/stay for the whole thing, join and leave at the beginnings and ends of each working session as they fit your schedule!*
- 9:30a.m. Welcome + introductions
- 9:40a.m. An introduction to GitHub - this is not meant to be an exhaustive intro of all functions, but just the basic to get you started with contributing to the website (i.e. repos, issues, PRs markdown) - *if anyone wants to teach & share here, please let me know!*
- 10:10a.m. break
- 10:20a.m. Introduction to the work to be done, break up into smaller teams(? - depends on number of participants), and introduction to the format of the day
- 10:30a.m. First working session (20 mins work, 5 mins reporting out, 5 mins break)
- 11:00a.m. Second working session
- 11:30a.m. Third working session (let see if we get to this!)

# Prerequisites - what do I need in order to join?
- **A GitHub account**! Sign up at https://github.com/signup

For the October 30 session:
- Access to Zoom - preferably version 5.3 or higher so you can sort yourselves into breakout rooms (;
- Access to audio and video, and the internet
- If you have some time beforehand - learning a bit of Markdown may be helpful: https://www.markdownguide.org/basic-syntax/

That's it! You don't have to know how to code, how to use GitHub, or anything.
 
# Questions? 
Please comment below, or email f [dot] tsang [at] tudelft [dot] nl.

**Comment by jurra** [comment]
I have some ideas for the website:
- The most difficult I find with regard to websites is how they support certain workflow or activity, who creates content and how is this organized. This includes considering: quantity and quality of content creators.
- Netlify cms can be integrated with github or gitlab. This allows for easy editing of contents, managing drafts before publishing etc. [Here are some examples](https://www.netlifycms.org/docs/start-with-a-template/). We could try this also on Friday :)
- It would be great to integrate the different initiatives within the open science website, so that our audience has a better idea of all the things happening. For instance the work groups, roles and actual people behind the program  This might include the carpentries, the role of data stewards, Data managers and RSEs, but also the 4TU repositories, and many other things.
- Describing what we do first, why we do it and then talk a bit about how we do it, the working groups and various initiatives, programs, projects would be great to capture.

**Comment by emmyft** [comment]
Thanks @jurra for reminding us to keep this discussion open (: 

Here's are my thoughts: 

- website workflows - this ties in with community governance. I've turned this into a separate issue to start the discussion (see #2 )
- netlify CMS - super idea! I've turned that into an issue in the website repo.
- working groups and different initiatives - definitely. i'll be very happy if the OSCD community have 5-6 of these by June next year, lets say (: What i'm tackling at the moment is the fact that the existing OSCD members are mostly interested in data (naturally - they were data champions ). To grow these other communities we'll need folks who are interested in other aspects of open science. It'd be great if the website can showcase the various initiatives and activities of OSCD (and I do think it can!), the question is: how do we diversify these initiatives (to effectively go beyond data and software) , such that folks who are not members, just browsing, don't feel like this is just about open data and software? or am i worrying too much and just have to accept that data and software are going to be our starting point... NOW #3
- "Describing what we do first, why we do it and then talk a bit about how we do it" - yes, super important! I've taken a stab at it, but this is not something I want to impose on the community. I want to make sure that community members are heard.... how do we do this effectively and inclusively? - also in #3

**Comment by junzis** [comment]
I have a few inputs for the discussion. (sorry that I won't be able to join the live session tomorrow..)

- Do we want to create a customized DNS for the website? Maybe something like **osc-delft.tudelft.nl** (not sure about the ownership), or **osc-delft.org**?

- Are we going to merge the data champion website to this website in the future? It might be nice to keep in mind before we start the developing.

- I actually prefer the plain GitHub/Jekyll/PR to any third-party CMS. In my opinion, CMS adds extract layer of complication...

**Comment by emmyft** [comment]
Thank you very much @junzis - I've included your input into the corresponding threads to help us work on them further :)
- DNS: https://github.com/osc-delft/osc-delft.github.io/issues/5
- CMS: https://github.com/osc-delft/osc-delft.github.io/issues/16

The data champions site - my idea is that this would _replace_ the current website.... except we'd need to sort out a new way to let people sign up, as the current sign up form for OSCD is still the data champions one hosted on the TU Delft website. (see https://github.com/osc-delft/osc-delft.github.io/issues/9) 

Does that sound logical...? is there a reason why we may want to keep the old site if all the content is transferred over? (maybe we should discuss this in another issue...)

**Comment by emmyft** [comment]
this is in the past, so will now close this (;
